src/classes/RD_RecurringDonations.cls

```diff
@@ -795,77 +766,391 @@ public class RD_RecurringDonations {
+                    Decimal remainingAmount = r.npe03__Amount__c - paidAmount;
+                    if (remainingInstallments > 0)
+                        installmentAmount = remainingAmount / remainingInstallments;
+                }
+            }
+
+            Integer installmentCounter = 1;
+            for (Opportunity o : RDOppMap.get(r.Id)) {
+                if (installmentCounter<=numInstallments) {
+                    if (o.isWon != true) {
+                        // fix close dates on existing opportunities
+                        o.CloseDate = nextDate;
+
+                        o.Amount = installmentAmount;
+                        if (r.npe03__Recurring_Donation_Campaign__c != null && rds.npe03__Add_Campaign_to_All_Opportunites__c) {
+                            o.CampaignId = r.npe03__Recurring_Donation_Campaign__c;
+                        }
+
+                        if (r.npe03__Organization__c != null) {
+                            o.AccountID = r.npe03__Organization__c;
+                        }
+
+                        //modify the name
+                        String oName = '';
+                        oName += r.npe03__Donor_Name__c;
+                        oName += ' ';
+                        oName += system.label.npe03.RecurringDonationPrefix;
+                        oName += ' (';
+                        oName += installmentCounter;
+                        oName += ' of ';
+                        oName += numInstallments;
+                        oName += ') ';
+                        oName += o.CloseDate.format();
+                        o.Name = oName;
+
+                        //evaluate any custom field mappings
+                        for (string s : customFieldMappingSettings.keySet()) {
+                            npe03__Custom_Field_Mapping__c cfm = customFieldMappingSettings.get(s);
+                            o.put(cfm.npe03__Opportunity_Field__c, r.get(cfm.npe03__Recurring_Donation_Field__c));
+                        }
+
+                        updateOppsList.add(o);
+
+                        nextDate = getNextDate(nextDate, r);
+                    }
+                    installmentCounter++;
+                } else {
+                    // extras get closed
+                    closeOppsList.add(o);
+                }
+            }
+
+            for (Integer i=RDOppMap.get(r.Id).size(); i<numInstallments; i++) {
+                Opportunity o = newOpportunityOnDate(nextDate,r,rdcounter);
+                insertOppsList.add(o);
+                nextDate = getNextDate(nextDate, r);
+            }
+
+        }
+
+        if (!insertOppsList.isEmpty()) {
+            Database.SaveResult[] lsr;
+            list<Database.SaveResult> badSRs = new list<Database.SaveResult>();
+            list<Opportunity> badOpps = new list<Opportunity>();
+
+            lsr = Database.insert(insertOppsList, false);
+
+            set<id> updateRecurringDonationSet = new set<id>();
```

Comment:
Same here -- error checking logic is below this loop that gets the Opp.RecurringDonation id.